pkg/loadbalancer/proxy.go

```diff
@@ -115,11 +139,11 @@ func generateConfig(service *v1.Service, nodes []*v1.Node) *proxyConfigData {
 	for _, ipFamily := range service.Spec.IPFamilies {
 		// TODO: support UDP
 		for _, port := range service.Spec.Ports {
-			if port.Protocol != v1.ProtocolTCP {
+			if port.Protocol != v1.ProtocolTCP && port.Protocol != v1.ProtocolUDP {
 				klog.Infof("service port protocol %s not supported", port.Protocol)
 				continue
 			}
-			key := fmt.Sprintf("%s_%d", string(ipFamily), port.Port)
+			key := fmt.Sprintf("%s_%d_%s", string(ipFamily), port.Port, port.Protocol)
```

Comment:
(It's inconsistent to cast `ipFamily` to `string` but not `port.Protocol`. You don't need to cast either of them.)

Reply:
how do you find these things :)